### PR TITLE
Store unversioned scripts in other directory

### PR DESCRIPTION
**Pull request by smusoke**
**What:** The directory that scripts with a `version_year` of `unversioned` are saved in - won't be the same as versioned scripts.

**Why:** Unversioned scripts were being skipped with the existing logic

## Links

- jira ticket: [FND-1666](https://codedotorg.atlassian.net/browse/FND-1666)

## Testing story

Ran `sync-in.rb` locally and no scripts were skipped

Before:
Screen Shot 2021-08-18 at 12 56 51 PM

After:
Screen Shot 2021-08-18 at 1 43 51 PM

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by Hamms — COMMENTED** [review]
This change looks good! Should definitely preserve consistency, but I have one nit about the implementation. I'd rather see this logic laid out in a way that makes the most sense given the new logic, rather than in a way that most easily fits into the existing layout.

I'm thinking something like 

```ruby
# in script model
def unversioned?
  version_year.blank? || version_year == 'unversioned'
end
```

```ruby
      script_i18n_directory =
        if ScriptConstants.unit_in_category?(:hoc, script.name)
          File.join(level_content_directory, "Hour of Code")
        elsif script.unversioned?
          File.join(level_content_directory, "other")
        else
          File.join(level_content_directory, script.version_year)
        end
```

**Comment by smusoke** [comment]
> This change looks good! Should definitely preserve consistency, but I have one nit about the implementation. I'd rather see this logic laid out in a way that makes the most sense given the new logic, rather than in a way that most easily fits into the existing layout.
> 
> I'm thinking something like
> 
> ```ruby
> # in script model
> def unversioned?
>   version_year.blank? || version_year == 'unversioned'
> end
> ```
> 
> ```ruby
>       script_i18n_directory =
>         if ScriptConstants.unit_in_category?(:hoc, script.name)
>           File.join(level_content_directory, "Hour of Code")
>         elsif script.unversioned?
>           File.join(level_content_directory, "other")
>         else
>           File.join(level_content_directory, script.version_year)
>         end
> ```

@Hamms I agree that's a better solution long term. Made the update in the latest commit.

File: dashboard/app/models/script.rb [hunk 1]
```diff
@@ -1812,6 +1812,10 @@ def course_title
     unit_group.try(:localized_title)
   end
 
+  def unversioned?
+    version_year.blank? || version_year == 'unversioned'
```
Comment (Hamms) [inline dashboard/app/models/script.rb]: I poked around a bit more, and it looks like we actually have a constant for this!
```suggestion
    version_year.blank? || version_year == CourseVersion::UNVERSIONED
```

**Review by Hamms — APPROVED** [review]
One final nit, then LGTM!

File: dashboard/app/models/script.rb (hunk 1 above)
Comment (smusoke) [reply dashboard/app/models/script.rb]: Ah thanks for finding this!

**Pull request merged**